### PR TITLE
Updates rest api generated files to preview2

### PR DESCRIPTION
**Pull request by brendankowitz**
Do not merge before S65 is in PPE

### CODE REVIEW

File: Microsoft.HealthVault.RestApi/Generated/ActionPlanObjectives.cs [hunk 1]
```diff
@@ -60,7 +60,7 @@ public ActionPlanObjectives(MicrosoftHealthVaultRestApi client)
         /// <param name='cancellationToken'>
         /// The cancellation token.
         /// </param>
-        /// <exception cref="MicrosoftKhronosCloudCommonErrorsErrorResponseException">
```
Comment (HATIMS) [inline Microsoft.HealthVault.RestApi/Generated/ActionPlanObjectives.cs]: >MicrosoftKhronosCloudCommonErrorsErrorResponseException [](start = 29, length = 55)

Does these changes happen every time the Rest client code is re-generated?

Comment (brendankowitz) [inline Microsoft.HealthVault.RestApi/Generated/ActionPlanObjectives.cs]: We corrected the long type names so "ErrorResponseException" is what it should be

---
In reply to: [121519936](https://github.com/Microsoft/healthvault-dotnetstandard-sdk/pull/54#discussion_r121519936) [](ancestors = 121519936)

File: Microsoft.HealthVault.RestApi/Generated/ActionPlanObjectives.cs [hunk 2]
```diff
@@ -195,24 +195,6 @@ public ActionPlanObjectives(MicrosoftHealthVaultRestApi client)
...
-                catch (JsonException ex)
-                {
-                    _httpRequest.Dispose();
-                    if (_httpResponse != null)
-                    {
-                        _httpResponse.Dispose();
-                    }
-                    throw new SerializationException("Unable to deserialize the response.", _responseContent, ex);
-                }
-            }
```
Comment (gcorvera) [inline Microsoft.HealthVault.RestApi/Generated/ActionPlanObjectives.cs]: Why remove this? 204 could still happen right? I wonder if we should try to merge all 20x responses into one.

File: Microsoft.HealthVault.RestApi/Generated/ActionPlanTasks.cs [hunk 3]
```diff
@@ -961,24 +961,6 @@ public ActionPlanTasks(MicrosoftHealthVaultRestApi client)
...
-                catch (JsonException ex)
-                {
-                    _httpRequest.Dispose();
-                    if (_httpResponse != null)
-                    {
-                        _httpResponse.Dispose();
-                    }
-                    throw new SerializationException("Unable to deserialize the response.", _responseContent, ex);
-                }
-            }
```
Comment (gcorvera) [inline Microsoft.HealthVault.RestApi/Generated/ActionPlanTasks.cs]: Same as in the other file, we still need this, right?

File: Microsoft.HealthVault.RestApi/Generated/GoalRecommendations.cs [hunk 4]
```diff
@@ -0,0 +1,818 @@
...
+    using System.IO;
+    using System.Net;
+    using System.Net.Http;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// GoalRecommendations operations.
+    /// </summary>
+    public partial class GoalRecommendations : IServiceOperations<MicrosoftHealthVaultRestApi>, IGoalRecommendations
```
Comment (gcorvera) [inline Microsoft.HealthVault.RestApi/Generated/GoalRecommendations.cs]: >GoalRecommendations [](start = 25, length = 19)

Brendan, check with Matt and Victor if this is something that will be out in 66

File: Microsoft.HealthVault.RestApi/Generated/Sleeps.cs [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+    using System.IO;
+    using System.Net;
+    using System.Net.Http;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Sleeps operations.
+    /// </summary>
+    public partial class Sleeps : IServiceOperations<MicrosoftHealthVaultRestApi>, ISleeps
```
Comment (gcorvera) [inline Microsoft.HealthVault.RestApi/Generated/Sleeps.cs]: >Sleeps [](start = 25, length = 6)

Talk with Justin about Sleep. Not sure if this is still ready

**Comment by gcorvera** [comment]
:clock1:

File: Microsoft.HealthVault.IntegrationTest/ActionPlanTests.cs [hunk 6]
```diff
@@ -119,10 +119,8 @@ private static ActionPlanTask CreateDailyWeightMeasurementActionPlanTask(string
                     }
                 },
                 CompletionType = "Frequency",
-                FrequencyTaskCompletionMetrics = new ActionPlanFrequencyTaskCompletionMetrics()
+                FrequencyTaskCompletionMetrics = new ActionPlanFrequencyTaskCompletionMetricsV2()
                 {
-                    ReminderState = "Off",
```
Comment (LiquidPT) [inline Microsoft.HealthVault.IntegrationTest/ActionPlanTests.cs]: This probably needs to get moved into a Schedule object in the Schedules array

File: Microsoft.HealthVault.RestApi/Generated/ActionPlanObjectives.cs (hunk 2 above)
Comment (LiquidPT) [inline Microsoft.HealthVault.RestApi/Generated/ActionPlanObjectives.cs]: 204 is the expected response on a delete

File: Microsoft.HealthVault.RestApi/Generated/GoalRecommendations.cs (hunk 4 above)
Comment (LiquidPT) [inline Microsoft.HealthVault.RestApi/Generated/GoalRecommendations.cs]: It's definitely available in 65. Was likely available in 64

**Comment by HATIMS** [comment]
:shipit:

**Comment by brendankowitz** [comment]
Included with https://github.com/Microsoft/healthvault-dotnetstandard-sdk/pull/59

**Pull request closed without merging**